### ISSUE TITLE
problem with collecting from particular Facebook pages

### ISSUE THREAD

**Issue by rjackland** [vosonlab/SocialMediaLab#29]
Attempting to collect from a particular Facebook page and while there appears to be public posts being collected, when it gets to the stage of "Collecting maximum of 1000 comments and likes for each post." there are numerous error messages:

Facebook API caused an unknown error while retrieving post ID,  [post ID] 
Condition:  Error in while (n.l < n.likes & length(content$data) > 0 & !is.null(url <- content$paging$`next`)) {: argument is of length zero

The resulting graph is empty, as is the csv file (dumped data frame).

This problem was originally reported by another user and I confirmed the problem.

Can still collect successfully from other Facebook pages.

**Comment by chainsawriot** [comment]
It probably is a problem introduces by v2.11 of graph API in which no user information will be returned anymore unless the access token is created by the page's owner.

https://developers.facebook.com/docs/graph-api/changelog/version2.11#gapi-90-pages

**Comment by rjackland** [comment]
Yes I confirm that this behaviour is a result of the v2.11 FB API which was introduced 7 Nov 2017 but some of the effects only became apparent in recent weeks due to:

"90-Day Breaking Changes — Changes and deprecations that will take effect 90 days after the version release date."

I just tried my test FB data collection which ran fine when I first tried it around 1st Jan but I can confirm that now all I get back is the posts on the page: none of the comments/likes.

So with v2.11, Facebook collection in SocialMediaLab is effectively broken, unless all you want is the posts by the page owner, or (as @chainsawriot noted above) the access token is created by the page owner.
